source/solvers/tracer.cc

```diff
@@ -530,6 +553,182 @@ Tracer<dim>::calculate_tracer_statistics()
+
+                      // We update the fields required by the diffusivity
+                      // model
+                      fields.clear();
+                      if (diffusivity_model->depends_on(field::levelset))
+                        {
+                          std::vector<double> levelset_values(n_q_points_face);
+                          fields.insert(std::pair<field, std::vector<double>>(
+                            field::levelset, n_q_points_face));
+                          face_quadrature_points =
+                            fe_face_values_tracer.get_quadrature_points();
+                          immersed_solid_signed_distance_function =
+                            this->multiphysics
+                              ->get_immersed_solid_signed_distance_function();
+                          this->immersed_solid_signed_distance_function
+                            ->value_list(face_quadrature_points,
+                                         levelset_values);
+                          set_field_vector(field::levelset,
+                                           levelset_values,
+                                           fields);
+                        }
+
+                      diffusivity_model->vector_value(fields,
+                                                      tracer_diffusivity);
+
+                      // Get fluid dynamics active cell iterator
+                      typename DoFHandler<dim>::active_cell_iterator cell_fd(
+                        &(*(this->triangulation)),
+                        cell->level(),
+                        cell->index(),
+                        dof_handler_fd);
+
+                      // Gather fluid dynamics information
+                      fe_face_values_fd.reinit(cell_fd, face);
+                      fe_face_values_fd[velocities].get_function_values(
+                        current_solution_fd, velocity_values);
+
+                      // Loop on the quadrature points
+                      for (unsigned int q = 0; q < n_q_points_face; q++)
+                        {
+                          Tensor<1, dim> normal_vector_tracer =
+                            -fe_face_values_tracer.normal_vector(q);
+
+                          tracer_flow_rate_vector[vector_index] +=
+                            (-tracer_diffusivity[q] * tracer_gradient[q] *
+                               normal_vector_tracer +
+                             tracer_values[q] * velocity_values[q] *
+                               normal_vector_tracer) *
+                            fe_face_values_tracer.JxW(q);
+                        } // end loop on quadrature points
+                    }     // end condition face at boundary
+                }         // end loop on faces
+            }             // end face is a boundary face
+        }                 // end condition cell at boundary
+    }                     // end loop on cells
```

Comment:
Why do you check the boundary_ids if you don't select a specific one?

```suggestion
                        } // end loop on quadrature points
                    }     // end condition face at boundary 
                }         // end condition face is a boundary face
            }             // end loop on faces
        }                 // end condition cell at boundary
    }                     // end loop on cells
```